File: llm_outputs/octocoder/GuruSamples/typescript/extract-method_isolate-switch_refactored.ts
class Order {
  //...

  calculateTotal(): number {
    let total = 0;
    for (const product of this.getProducts()) {
      total += product.quantity * product.price;
    }

    // Apply regional discounts.
    switch (this.user.getCountry()) {
      case "US":
        total *= 0.85;
        break;
      case "RU":
        total *= 0.75;
        break;
      case "CN":
        total *= 0.9;
        break;
      //...
    }

    return total;
  }
}

File: results/llm_outputs/codellama7b/GuruSamples/java/extract-method_isolate-switch_refactored.java
class Order {
  //...

  public double calculateTotal() {
    double total = 0;
    for (Product product : getProducts()) {
      total += product.quantity * product.price;
    }

    // Apply regional discounts.
    total *= getDiscountFactor(user.getCountry());

    return total;
  }

  private double getDiscountFactor(String country) {
    switch (country) {
      case "US": return 0.85;
      case "RU": return 0.75;
      case "CN": return 0.9;
      //...
      default: return 1.0;
    }
  }
}
